qcelemental/models/molecule.py

```diff
@@ -753,17 +758,77 @@ def get_molecular_formula(self):
         ClH
 
         """
-        count = collections.Counter(x.title() for x in self.symbols)
+        return self._formula_from_symbols(self.symbols, order)
+
+    @staticmethod
+    def _formula_from_symbols(symbols: List[str], order: str = "alphabetical") -> str:
+        """
+        Returns the molecular formula for a list of symbols.
+
+        Pararmeters
+        -----------
+        symbols: List[str]
+            List of chemical symbols
+        order: str, optional
+            sorting order of the formula. Valid choices are "alphabetical" and "hill".
+
+        Returns
+        -------
+        str
+            The molecular formula
+        """
+
+        supported_orders = ["alphabetical", "hill"]
+        if order.lower() not in supported_orders:
+            raise ValueError(f"Unsupported molecular formula order: {order}. Supported orders are f{supported_orders}.")
+        count = collections.Counter(x.title() for x in symbols)
+        element_order = sorted(count.keys())
+
+        if order.lower() == "hill" and "C" in element_order:
+            if "H" in element_order:
+                element_order.insert(0, element_order.pop(element_order.index("H")))
+            element_order.insert(0, element_order.pop(element_order.index("C")))
 
         ret = []
-        for k in sorted(count.keys()):
+        for k in element_order:
             c = count[k]
             ret.append(k)
             if c > 1:
                 ret.append(str(c))
 
         return "".join(ret)
 
+    @staticmethod
+    def order_molecular_formula(formula: str, order: str = "alphabetical") -> str:
```

Comment:
I worry a bit about function explosion with items like this on the Molecule class. Can you think of another place to put it?

Reply:
molutil???

Reply:
Yea, would that be ok?

Reply:
Yep!